app/message-table.tsx

```diff
@@ -17,19 +20,22 @@ import {
   TableHeader,
   TableRow,
 } from "@/components/ui/table";
-
-import { useState } from "react";
+import { Checkbox } from "@/components/ui/checkbox";
 
 interface MessageTableProps<TData, TValue> {
```

Comment:
This function has gotten large enough with the JSX deeply nested and indented enough that it feels hard to reason about.  I'd like to suggest pulling out the `<TableHead>` JSX invocation into its own functional component (could be in the same file, though). I also wonder if the first `TableBody` invocation wants to be in its own component as well, but I'm more on the fence about that. As long as you're testing the high-level component, I don't this/these internal implementation components would need additional unit testing.